src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,733 @@
+            }
+        }
+    }
+
+    /**
+     * @notice Get the number of queued dealloations for the given (operator, strategy, operatorSetKey) tuple
+     * @param operator address to get queued deallocations for
+     * @param strategy the strategy to get queued deallocations for
+     * @param operatorSetKey the encoded operatorSet to get queued deallocations for
+     */
+    function _getNumQueuedDeallocations(
+        address operator,
+        IStrategy strategy,
+        bytes32 operatorSetKey
+    ) internal view returns (uint256) {
+        uint256 numQueuedDeallocations;
+
+        uint256 length = _queuedDeallocationIndices[operator][strategy][operatorSetKey].length;
+        for (uint256 i = length; i > 0; --i) {
+            // index of pendingFreeMagnitude/deallocation to check for slashing
+            uint256 index = _queuedDeallocationIndices[operator][strategy][operatorSetKey][i - 1];
+
+            // If completableTimestamp is greater than completeUntilTimestamp, break
+            if (block.timestamp < _pendingFreeMagnitude[operator][strategy][index].completableTimestamp) {
+                ++numQueuedDeallocations;
+            } else {
+                break;
+            }
+        }
+
+        return numQueuedDeallocations;
+    }
+
+    /// @dev gets the latest total magnitude or overwrites it if it is not set
+    function _getLatestTotalMagnitude(address operator, IStrategy strategy) internal returns (uint64) {
+        (bool exists,, uint224 totalMagnitude) = _totalMagnitudeUpdate[operator][strategy].latestSnapshot();
+        if (!exists) {
+            totalMagnitude = WAD;
+            _totalMagnitudeUpdate[operator][strategy].push({key: uint32(block.timestamp), value: totalMagnitude});
+            operatorMagnitudeInfo[operator][strategy].freeMagnitude = WAD;
+        }
+
+        return uint64(totalMagnitude);
+    }
+
+    /// @dev gets the pending free magnitude available to add by completing numToComplete pending deallocations
+    /// and returns the next index to start from if completed.
+    function _getPendingFreeMagnitude(
+        address operator,
+        IStrategy strategy,
+        uint16 numToComplete,
+        uint192 nextIndex
+    ) internal view returns (uint64 freeMagnitudeToAdd, uint192 completed) {
+        uint256 pendingFreeMagnitudeLength = _pendingFreeMagnitude[operator][strategy].length;
+        freeMagnitudeToAdd = 0;
+        while (nextIndex < pendingFreeMagnitudeLength && completed < numToComplete) {
+            PendingFreeMagnitude memory pendingFreeMagnitude = _pendingFreeMagnitude[operator][strategy][nextIndex];
+            // pendingFreeMagnitude is ordered by completableTimestamp. If we reach one that is not completable yet, then break
+            // loop until completableTimestamp is < block.timestamp
+            if (block.timestamp < pendingFreeMagnitude.completableTimestamp) {
```

Comment:
nit: the comment is opposite to the check here. the condition appears incorrect, the comment should be corrected (`<` => `>`)